test/Test/Toml/Parsing/Unit.hs

```diff
@@ -0,0 +1,146 @@
+{-# LANGUAGE FlexibleContexts #-}
+
+module Test.Toml.Parsing.Unit where
+
+import Test.Tasty.Hspec (Spec, context, describe, it)
+
+import Data.Text (singleton)
+import Test.Hspec.Megaparsec
+import Text.Megaparsec (parse)
+
+import Toml.Parser (arrayP, boolP, integerP)
+import Toml.Type (UValue (..))
+
+spec_Parser :: Spec
+spec_Parser = do
+  let parseX p given expected = parse p "" given `shouldParse` expected
+      failOn p given          = parse p "" `shouldFailOn` given
+
+  let parseArray  = parseX arrayP
+      parseBool   = parseX boolP
+      parseInt    = parseX integerP
+      arrayFailOn = failOn arrayP
+      boolFailOn  = failOn boolP
+
+  describe "arrayP" $ do
+    it "can parse arrays" $ do
+      parseArray "[]" []
+      parseArray "[1]" [UInt 1]
+      parseArray "[1, 2, 3]" [UInt 1, UInt 2, UInt 3]
+      parseArray "[1.2, 2.3, 3.4]" [UFloat 1.2, UFloat 2.3, UFloat 3.4]
+      parseArray "['x', 'y']" [UString (singleton 'x'), UString (singleton 'y')]
```

Comment:
`singleton 'x'` is just `"x"`.

Reply:
@ChShersh Using `"x"` instead of `singleton 'x'` won't pass the test, because `UString` takes an argument of type `Text` and not `String`.
https://github.com/kowainik/tomland/blob/9d44db6f56088c38d76a443b74defbd5ebd5245b/src/Toml/Type.hs#L241